### PR TITLE
docs(jsdoc): add examples for endWith operator

### PR DESCRIPTION
**Pull request by moniuch**
**Description:**
Add jsdoc examples for `endWith` operator.

### CODE REVIEW

**Comment by moniuch** [comment]
What do we do when adding comments decreases the coverage?

**Comment by JWO719** [comment]
Hi @moniuch . I think it isn't a problem with your code. Maybe the tool had some problems. One of my prs had exactly the same problem. I'll take a look at it this afternoon, maybe I find something. Don't panic :D

**Comment by moniuch** [comment]
Just `KEEP CALM AND CONTRIBUTE`  :)

@JWO719 Would you mind if I asked you for help?... After I `npm install`-ed, I want to make sure everything works ok locally for me. Do we have any "take-off" checklist to run? I really want to contribute with some real code now (first operator, even if it is ends up not being accepted in PR).

I feel like this kind of checklist could make it to CONTRIBUTING.md.

**Comment by JWO719** [comment]
Of course you can ask me anytime for help :D But I actually don't understand what you are looking for?

**Comment by JWO719** [comment]
Ok now I get it. IMO it makes sense to add stuff like this to the CONTRIBUTING.md. So in general I would suggest running all tests makes sense due to possible "side effects". And in general tests should be executed with `npm run test`. Did that helped you? But anyhow updating the CONTIBUTING.md by making a proposal is definetly a good idea.

**Comment by moniuch** [comment]
Thanks, will give it a shot and see how it goes. The main point is - how do I test my new operator locally in a playground, just like I can test published release on stackblitz? Is there a nice flow for that?
I sure can create a ticket for updating CONTRIBUTING.md, but I don't think I would be the right person to take it :)

**Comment by JWO719** [comment]
maybe @benlesh or @cartant have some tipps for you

**Comment by cartant** [comment]
@moniuch If you are looking at writing an operator, you should start with a user-land operator. That is, you should write an operator that is not part of the RxJS package.

Some information from OJ:

> Our general recommendation is if implementation is trivial / and doesn't require modification in core, we do recommend for user-land first. We want to gather usecases around how much common it is before we include into core, since adding new api surface is easy but deprecating / changing behavior is exceptionally hard, as well as we don't want to core api surface it too broad (it is alreay, and we are trying to shrink it down).

Regarding a playground, it's up to you. All you really need is the UMD bundle and a browser (see the operator in [this Stack Overflow answer](https://stackoverflow.com/a/50654079/6680611).

It should be pretty easy to use either StackBlitz or CodeSandbox (via the vanilla starting point) to get an in-browser, TypeScript environment happening. Just import RxJS and get started.

**Comment by moniuch** [comment]
@cartant Thanks for the info. So, by userland you mean just any type of fiddle, not any specific platform. Would this be ok? https://stackblitz.com/edit/ts-rxjs-some-userland?file=index.ts

**Comment by cartant** [comment]
@moniuch A user-land operator is just an operator that's separate from the RxJS package.

For example, my [`rxjs-etc`](https://github.com/cartant/rxjs-etc) package has a bunch of operators in it. They are user-land operators, as they're not part of RxJS proper.

**Comment by moniuch** [comment]
@cartant Oh, you got an entire repo of those! Nice, thanks - this gives me a much better idea of how to go about it.

File: src/internal/operators/endWith.ts [hunk 1]
```diff
@@ -21,6 +21,44 @@ export function endWith<T>(...array: Array<T | SchedulerLike>): MonoTypeOperator
  * Returns an Observable that emits the items you specify as arguments after it finishes emitting
  * items emitted by the source Observable.
  *
+ * ## Examples
+ * ### Emits a value before completing (appends an emission to a series of emissions).
+ *
+ * ```javascript
+ * from(['hi', 'how are you?', 'sorry, I have to go now']).pipe(
```
Comment (benlesh) [inline src/internal/operators/endWith.ts]: I think use `of` here instead of `from`, makes the code a *little* simpler.

**Comment by moniuch** [comment]
@benlesh Agreed - this is exactly what I did for `ignoreElements` - somehow didn't think of applying similar changes to `endWith`. Fixed.

File: src/internal/operators/endWith.ts [hunk 2]
```diff
@@ -21,11 +21,11 @@ export function endWith<T>(...array: Array<T | SchedulerLike>): MonoTypeOperator
  * Returns an Observable that emits the items you specify as arguments after it finishes emitting
  * items emitted by the source Observable.
  *
- * ## Examples
- * ### Emits a value before completing (appends an emission to a series of emissions).
+ * ## Example
+ * ### After the source observable completes, appends an emission and then completes too.
```
Comment (benlesh) [inline src/internal/operators/endWith.ts]: one or more emissions, actually...  but the most common use case is one. :)

**Comment by benlesh** [comment]
Thanks, @moniuch

**Pull request merged**